doc/Type/IO/Socket/Async.pod6

```diff
@@ -12,13 +12,26 @@ server and the client side.
 Here is the equivalent example to that in L<IO::Socket::INET> of a simple
 echo server that listens on port 3333:
 
-=begin code
-use v6.c;
-
-react {
-    whenever IO::Socket::Async.listen('localhost', 3333) -> $conn {
-        whenever $conn.Supply(:bin) -> $buf {
-            await $conn.write: $buf
+=begin code :skip-test<IO>
+loop {
+    react {
+        whenever IO::Socket::Async.listen('0.0.0.0', 8080) -> $conn {
+            react { 
+                whenever $conn.Supply.lines {
+                    say .Str;
+
+                    { await $conn.print(.Str ~ "\n") } for 
+                    "HTTP/1.1 200 OK",
+                    "Content-Type: text/html; charset=UTF-8",
+                    "Content-Encoding: UTF-8",
+                    "",
+                    "<html><body>Hello World!</body></html>";
+
+                    $conn.close;
```

Comment:
Generally, having the react/whenever block suggests that you might have several whenevers, but in this particular case we are showing a less than awesome precedent of ``await``-ing inside the whenever and therefore not letting other whenevers do the job (there are none in this case, but as an example it is less than awesome).